hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -205,11 +205,17 @@ protected void onRemoveError(K key, R record, boolean removed, Throwable error)
 
     @SuppressWarnings("unused")
     protected void onExpire(K key, R record) {
+        if (!canUpdateStats(record)) {
+            return;
+        }
         nearCacheStats.incrementExpirations();
     }
 
     @Override
     public void onEvict(KS key, R record, boolean wasExpired) {
+        if (!canUpdateStats(record)) {
```

Comment:
If we want to put these checks into both of the places, maybe we can add a check to overridden `onEvict` method in `HDNearCacheRecordStoreImpl` as well.

Reply:
EE side PR https://github.com/hazelcast/hazelcast-enterprise/pull/5913